docs/reference/how-to/search-speed.asciidoc

```diff
@@ -0,0 +1,183 @@
+queries hundreds of times slower. So if the same questions can be answered without
+joins by denormalizing documents, significant speedups can be expected.
+
+=== Pre-index data
+
+You should leverage patterns in your queries to optimize the way data is indexed.
+For instance, if all your documents have a `price` field and most queries run
+<<search-aggregations-bucket-range-aggregation,`range`>> aggregations on a fixed
+list of ranges, you could make this aggregation faster by pre-indexing the ranges
+into the index and using a <<search-aggregations-bucket-terms-aggregation,`terms`>>
+aggregations.
+
+For instance, if documents look like:
+
+[source,js]
+--------------------------------------------------
+PUT index/type/1
+{
+  "designation": "spoon",
+  "price": 13
+}
+--------------------------------------------------
+// CONSOLE
+// TEST
+
+and search requests look like:
+
+[source,js]
+--------------------------------------------------
+GET index/_search
+{
+  "aggs": {
+    "price_ranges": {
+      "range": {
+        "field": "price",
+        "ranges": [
+          { "to": 10 },
+          { "from": 10, "to": 100 },
+          { "from": 100 }
+        ]
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+Then documents could be enriched by a `price_range` field at index time, which
+should be mapped as a <<keyword,`keyword`>>:
+
+[source,js]
+--------------------------------------------------
+PUT index
+{
+  "mappings": {
+    "type": {
+      "properties": {
+        "price_range": {
+          "type": "keyword"
+        }
+      }
+    }
+  }
+}
+
+PUT index/type/1
+{
+  "designation": "spoon",
+  "price": 13,
+  "price_range": "10-100"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST
+
+And then search requests could aggregate this new field rather than running a
+`range` aggregation on the `price` field.
+
+[source,js]
+--------------------------------------------------
+GET index/_search
+{
+  "aggs": {
+    "price_ranges": {
+      "terms": {
+        "field": "price_range"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+=== Mappings
+
+The fact that some data is numeric does not mean it should always be mapped as a
+<<number,numeric field>>. Typically, fields storing identifiers might benefit
+from being mapped as <<keyword,`keyword`>> rather than `integer` or `long`.
+
+=== Avoid scripts
+
+In general, scripts should be avoided. If they are absolutely needed, you
+should prefer the `painless` and `expressions` engines.
+
+=== Force-merge read-only indices
+
+Indices that are read-only would benefit from being
+<<indices-forcemerge,merged down to a single segment>>. This is typically the
+case with time-based indices: only the index for the current time frame is
+getting new documents while older indices are read-only.
+
```

Comment:
Perhaps add "Don't force-merge indices that are still being written to -- leave merging to the background merge process."